pkg/chunk/chunk_store.go

```diff
@@ -93,22 +93,34 @@ func (c *store) Stop() {
 
 // Put implements ChunkStore
 func (c *store) Put(ctx context.Context, chunks []Chunk) error {
+	for _, chunk := range chunks {
+		if err := c.PutOne(ctx, chunk.From, chunk.Through, chunk); err != nil {
```

Comment:
This seems like rather a large change in performance characteristics, if you have a number of chunks to write.